src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Organization.ps1

```diff
@@ -401,21 +401,119 @@ class Organization: ADOSVTBase
-                {               
-                    $controlResult.AddMessage("No. of installed extensions: " + $extCount);
+                {   
+                    if([AzSKRoot]::IsDetailedScanRequired -eq $true)
+                    {
+                        # find inactive extensions
+                        $date = Get-Date
+                        $ExtensionsLastUpdatedInYears=$this.ControlSettings.Organization.ExtensionsLastUpdatedInYears;
+                        $thresholddate = $date.AddYears(-$ExtensionsLastUpdatedInYears)
+                        $staleExtensionList = $extensionList | Where-Object {([datetime] $_.lastPublished) -lt $thresholddate}
+                        $controlResult.AddMessage("`nNo. of extensions not have been updated since past $ExtensionsLastUpdatedInYears years : "+ $staleExtensionList.count)                        
+                        $display= $staleExtensionList|Format-Table -Property  @{name="ExtensionName";expression={$_.extensionName}},@{name="PublisherName";expression={$_.publisherName}} | Out-String
+                        $controlResult.AddMessage($display) 
+
+                        # display extensions with critical scopes
+                        $ExtensionCriticalScopes=$this.ControlSettings.Organization.ExtensionCriticalScopes;
+                        $ExtensionListWithCriticalScopes = $extensionList | Where-Object {$ExtensionCriticalScopes -contains $_.scopes }
+                        $controlResult.AddMessage("`nNo. of extensions have critical access permissions : "+ $ExtensionListWithCriticalScopes.count)                        
+                        $display= $ExtensionListWithCriticalScopes|Format-Table -Property  @{name="ExtensionName";expression={$_.extensionName}},@{name="Scope";expression={$_.scopes}} | Out-String
+                        $controlResult.AddMessage($display) 
+
+                        # Avoid extensions  with 'DevTest', 'Demo', 'Preview', 'Deprecated' in names
+                        $ExtensionListWithInvalidExtensionNames=@()
+                        $InvalidExtensionNames=$this.ControlSettings.Organization.InvalidExtensionNames;
+                        for($i=0;$i -lt $extensionList.count;$i++)
+                        {
+                            for($j=0;$j -lt $InvalidExtensionNames.Count;$j++)
+                            {
+                                if($extensionList[$i] -match $InvalidExtensionNames[$j])
+                                {
+                                    [System.Array] $ExtensionListWithInvalidExtensionNames = $extensionList[$i]
+                                }
+                            }
+                        }                        
+                        if($ExtensionListWithInvalidExtensionNames.count -gt 0)
+                        {
+                        $controlResult.AddMessage("`nExtensions having name as of test environment : "+ $ExtensionListWithInvalidExtensionNames.count)
```

Comment:
No. of extensions that have name belonging to non-production enevironment:

Reply:
Updated the message